packages/server/src/users/users.service.ts

```diff
@@ -1,13 +1,29 @@
 import { Injectable } from '@nestjs/common';
-import { Prisma, User } from '@prisma/client';
 import { PrismaService } from '../prisma/prisma.service';
 import * as argon2 from 'argon2';
 import { addHours, isFuture } from 'date-fns';
+import { Prisma, User } from '@prisma/client';
 
 @Injectable()
-export class UserService {
+export class UsersService {
   constructor(private prisma: PrismaService) {}
 
+  async findOneUsername(username: string): Promise<any> {
+    return await this.prisma.user.findUnique({
+      where: {
+        username: username
+      }
+    });
+  }
+
+  async findOneEmail(email: string): Promise<any> {
+    return await this.prisma.user.findUnique({
+      where: {
+        email: email
+      }
+    });
+  }
+
```

Comment:
Maybe remove these two duplicated functions and call findUserByUsername and findUserByEmail instead

Reply:
@gregfrasco should users have to be part of a project to be able to authenticate? based on our conversation, it would seem that users should provide a `project_id` when they login. If so, then I need to change the `LocalStrategy` to use a custom parameter - need to figure that part out. Then I can remove those redundant functions and use Wenhan's suggested functions.

Reply:
greg said that `project_id` should be passed in. but i can't find examples or documentation supporting custom configurations. seems like the only supported options are `usernameField`, `passwordField` and `session`:
https://github.com/jaredhanson/passport-local

Reply:
Does login has anything to do with passport?

Reply:
Oh also after I removed `@unique` for `username` and `email` from the schema, I'm getting 
`Type '{ username: string; }' is not assignable to type 'UserWhereUniqueInput'.
  Object literal may only specify known properties, and 'username' does not exist in type 'UserWhereUniqueInput'.`

This occurs because 
```
prisma.user.findUnique({
      where: {
        username: username
      }
    })
```
expects that the schema enforces uniqueness on `username`.

It seems like I can't pass in unique configurations to passport and therefore can't pass in `project_id`.  Kinda stuck on this.